receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper_test.go

```diff
@@ -128,6 +129,225 @@ func TestScrape(t *testing.T) {
+		metricsConfig       metadata.MetricsSettings
+		expectedMetricCount int
+		times               bool
+		utilization         bool
+	}
+
+	testCases := []testCase{
+		{
+			name:                "Standard",
+			metricsConfig:       metadata.DefaultMetricsSettings(),
+			expectedMetricCount: 2,
+			times:               true,
+			utilization:         true,
+		},
+		{
+			name:                "SystemCPUTime metric is disabled ",
+			times:               false,
+			utilization:         true,
+			expectedMetricCount: 1,
+		},
+		{
+			name:                "SystemCPUUtilization metric is disabled ",
+			times:               true,
+			utilization:         false,
+			expectedMetricCount: 1,
+		},
+		{
+			name:                "all metrics are disabled ",
+			times:               false,
+			utilization:         false,
+			expectedMetricCount: 0,
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			settings := test.metricsConfig
+			if test.metricsConfig == (metadata.MetricsSettings{}) {
+				settings = metadata.MetricsSettings{
+					SystemCPUTime: metadata.MetricSettings{
+						Enabled: test.times,
+					},
+					SystemCPUUtilization: metadata.MetricSettings{
+						Enabled: test.utilization,
+					},
+				}
+			}
+
+			scraper := newCPUScraper(context.Background(), &Config{Metrics: settings})
+			err := scraper.start(context.Background(), componenttest.NewNopHost())
+			require.NoError(t, err, "Failed to initialize cpu scraper: %v", err)
+
+			_, err = scraper.scrape(context.Background())
+			require.NoError(t, err, "Failed to scrape metrics: %v", err)
+			//2nd scrape will trigger utilization metrics calculation
+			md, err := scraper.scrape(context.Background())
+			require.NoError(t, err, "Failed to scrape metrics: %v", err)
+
+			assert.Equal(t, test.expectedMetricCount, md.MetricCount())
+			metrics := md.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics()
+			internal.AssertSameTimeStampForAllMetrics(t, metrics)
+			if test.times && test.utilization {
+				timesMetrics := metrics.At(0)
+				assertCPUMetricValid(t, timesMetrics, 0)
+				if runtime.GOOS == "linux" {
+					assertCPUMetricHasLinuxSpecificStateLabels(t, timesMetrics)
+				}
+
+				utilizationMetrics := metrics.At(1)
+				assertCPUUtilizationMetricValid(t, utilizationMetrics, 0)
+				if runtime.GOOS == "linux" {
+					assertCPUUtilizationMetricHasLinuxSpecificStateLabels(t, utilizationMetrics)
+				}
+			} else if test.times {
+				timesMetrics := metrics.At(0)
+				assertCPUMetricValid(t, timesMetrics, 0)
+				if runtime.GOOS == "linux" {
+					assertCPUMetricHasLinuxSpecificStateLabels(t, timesMetrics)
+				}
+			} else if test.utilization {
+				utilizationMetrics := metrics.At(0)
+				assertCPUUtilizationMetricValid(t, utilizationMetrics, 0)
+				if runtime.GOOS == "linux" {
+					assertCPUUtilizationMetricHasLinuxSpecificStateLabels(t, utilizationMetrics)
+				}
+			}
```

Comment:
The first condition seems redundant if we have next two as independent conditions
```
if {}
if {}
```

Reply:
👍 I'll add the utilization index in metrics to the test case